model/jpa/src/main/java/org/keycloak/models/jpa/RealmAdapter.java

```diff
@@ -1287,25 +1291,31 @@ public void addIdentityProvider(IdentityProviderModel identityProvider) {
         } else {
             entity.setInternalId(identityProvider.getInternalId());
         }
+        fillEntity(identityProvider, entity);
+        entity.setProviderId(identityProvider.getProviderId());
+
+        realm.addIdentityProvider(entity);
+
+        identityProvider.setInternalId(entity.getInternalId());
+
+        em.persist(entity);
+        em.flush();
+    }
+
+    private void fillEntity(IdentityProviderModel identityProvider, IdentityProviderEntity entity) {
```

Comment:
Not super keen on this refactoring - as it makes it more costly to review this. @mposolda did you review this properly?

Reply:
This was just a trivial extraction, but since I'm removing the changes in RealmAdapter, I'll also remove the refactoring.

Reply:
@stianst I think that this PR doesn't need to change anything in the RealmAdapter class at all if some of my other comments are applied - this means using just field from "config" map instead of adding additional field into the IdentityProviderModel / IdentityProviderMapperModel classes.

Reply:
Sounds good